Add `ydb sql` documentation and update links to use new command where possible (#15262)
ydb-platform/ydb: ydb/docs/en/core/reference/ydb-cli/sql.md

# Query execution

> You can use the `{{ ydb-cli }} sql` subcommand to execute any SQL query. The query can be of any type (DDL, DML e.t.c.) and can consist of several sub-queries. The `{{ ydb-cli }} sql` subcommand establishes a streaming connection and retrieves data through it. With the in-stream query execution, no limit is imposed on the amount of data read.

Reviewer: IIRC, there was a technical debt of some kinds of queries that can't be executed via query service. Has it been resolved, and there are indeed no exceptions?

> You can use the `{{ ydb-cli }} sql` subcommand to execute any SQL query. The query can be of any type (DDL, DML e.t.c.) and can consist of several sub-queries. The `{{ ydb-cli }} sql` subcommand establishes a streaming connection and retrieves data through it. With the in-stream query execution, no limit is imposed on the amount of data read.

Reviewer: I understand why you are focusing so much on reading large amounts of data, but for new users, other kinds of operations could be equally important. For example, how about writing large amounts of data? How does all this interact with transactions?

Let's make a broader intro about what it can do but without too deep technical details.

Author: This command can write data at some rate, but is not suited for that. There are probably other commands like bulkUpsert that would  upload large amounts of data better. But I'll mention this command can write data

> You can use the `{{ ydb-cli }} sql` subcommand to execute any SQL query. The query can be of any type (DDL, DML e.t.c.) and can consist of several sub-queries. The `{{ ydb-cli }} sql` subcommand establishes a streaming connection and retrieves data through it. With the in-stream query execution, no limit is imposed on the amount of data read.

Reviewer: Let's be careful with "any SQL". For instance, afaik we have three SQL dialects, and this article covers only one of them. Let's at least mention each one and link to more detailed reference materials.

Author: We have `--syntax` option hidden at the moment, so I'll just remove `any`

> # Query execution
> 
> You can use the `{{ ydb-cli }} sql` subcommand to execute any SQL query. The query can be of any type (DDL, DML e.t.c.) and can consist of several sub-queries. The `{{ ydb-cli }} sql` subcommand establishes a streaming connection and retrieves data through it. With the in-stream query execution, no limit is imposed on the amount of data read.
> 
> General format of the command:

Reviewer: ```suggestion
# Query execution

You can use the `{{ ydb-cli }} sql` subcommand to execute any SQL query. The query can be of any type (DDL, DML, etc.) and can consist of several subqueries. The `{{ ydb-cli }} sql` subcommand establishes a streaming connection and retrieves data through it. With in-stream query execution, no limit is imposed on the amount of data read.

General format of the command:
```

```bash
{{ ydb-cli }} [global options...] sql [options...]
```

> * `global options`: [Global parameters](commands/global-options.md).
> * `options`: [Parameters of the subcommand](#options).
> 
> View the description of this command by calling it with `--help` option:

Reviewer: ```suggestion
* `global options`: [Global parameters](commands/global-options.md).
* `options`: [Subcommand parameters](#options).

View the description of this command by calling it with `--help` option:
```

```bash
{{ ydb-cli }} sql --help
```

## Parameters of the subcommand {#options}

#|
|| Name | Description ||
|| `-h`, `--help` | Print general usage help ||
|| `-hh` | Print complete usage help. Prints some specific options that are not printed with `--help` ||
|| `-s`, `--script` | Script (query) text to execute. ||

> || `-f`, `--file` | Path to a file with query text to execute. Path `-` means reading query text from `stdin`. ||

Reviewer: I'd highlight that `-` is probably incompatible with passing parameters via `stdin`.

> || Name | Description ||
> || `-h`, `--help` | Print general usage help ||
> || `-hh` | Print complete usage help. Prints some specific options that are not printed with `--help` ||
> || `-s`, `--script` | Script (query) text to execute. ||
> || `-f`, `--file` | Path to a file with query text to execute. Path `-` means reading query text from `stdin`. ||
> || `--stats` | Statistics mode.<br/>Available options:<br/>

Reviewer: ```suggestion
|| Name | Description ||
|| `-h`, `--help` | Print general usage help. ||
|| `-hh` | Print complete usage help, including specific options not shown with `--help`. ||
|| `-s`, `--script` | Script (query) text to execute. ||
|| `-f`, `--file` | Path to a file containing the query text to execute. The path `-` means reading the query text from `stdin`. ||
|| `--stats` | Statistics mode.<br/>Available options:<br/>
```

<ul>

> <li>`none` (default): Do not collect.</li>
> <li>`basic`: Collect aggregated statistics for updates and deletes per table.</li>
> <li>`full`: Add execution statistics and plan on top of `basic`.</li>
> <li>`profile`: Collect detailed execution statistics including statistics for individual tasks and channels.</li>

Reviewer: ```suggestion
<li>`none` (default): Do not collect statistics.</li>
<li>`basic`: Collect aggregated statistics for updates and deletes per table.</li>
<li>`full`: Include execution statistics and plan in addition to `basic`.</li>
<li>`profile`: Collect detailed execution statistics, including statistics for individual tasks and channels.</li>
```

</ul> ||
|| `--explain` | Execute explain request for the query. Shows query logical plan. The query is not actually executed, thus does not affect database data. ||

> || `--explain-ast` | Same as `--explain`, but in addition to the query logical plan, you an AST (abstract syntax tree) is printed. The AST section contains a representation in the internal miniKQL language. ||

Reviewer: Make miniKQL to be a link to the glossary (add an entry if it's missing). Make AST to be the link to Wikipedia.

> || `--explain` | Execute explain request for the query. Shows query logical plan. The query is not actually executed, thus does not affect database data. ||
> || `--explain-ast` | Same as `--explain`, but in addition to the query logical plan, you an AST (abstract syntax tree) is printed. The AST section contains a representation in the internal miniKQL language. ||
> || `--explain-analyze` | Execute query in `EXPLAIN ANALYZE` mode. Shows query execution plan. Query results are ignored.<br/>**Important note: The query is actually executed, so any changes will be applied to the database**. ||
> || `--format` | Output format.<br/>Available options:

Reviewer: ```suggestion
|| `--explain` | Execute an explain request for the query. Displays the query's logical plan. The query is not actually executed and does not affect database data. ||
|| `--explain-ast` | Same as `--explain`, but in addition to the query's logical plan, an abstract syntax tree (AST) is printed. The AST section contains a representation in the internal miniKQL language. ||
|| `--explain-analyze` | Execute the query in `EXPLAIN ANALYZE` mode. Displays the query execution plan. Query results are ignored.<br/>**Important note: The query is actually executed, so any changes will be applied to the database**. ||
|| `--format` | Output format.<br/>Available options:
```

{% include notitle [format](./_includes/result_format_common.md) %}

{% include notitle [format](./_includes/result_format_csv_tsv.md) %}

||
|#

### Working with parameterized queries {#parameterized-query}

> A brief help is provided below. For a detailed description with examples, see [{#T}](parameterized-query-execution.md).

Reviewer: The text below feels like a copy-paste from the neighboring article. Let's just drop it, as we are already providing a link to it.
